stdlib/public/core/Random.swift

```diff
@@ -160,3 +160,9 @@ public struct SystemRandomNumberGenerator: RandomNumberGenerator {
     return random
   }
 }
+
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+@_silgen_name("arc4random_buf")
+@usableFromInline
+internal func swift_stdlib_random(_: inout UInt64, _: Int)
```

Comment:
```suggestion
internal func swift_stdlib_random(_: UnsafeMutableRawPointer, _: Int)
```

The [validation-test](https://github.com/apple/swift/blob/e9c8bb7d62fcf7f97b1a06613b31fc16eef87976/validation-test/stdlib/Random.swift#L12-L17) uses `swift_stdlib_random` with buffers larger than a `UInt64`.

Reply:
Also `arc4random_buf` uses the C calling conv.

Reply:
And you can't use `@_silgen_name` to call arbitrary C things regardless, because they'll collide with the real declarations when someone imports them.

Reply:
If `swift_stdlib_random` (for `__APPLE__` platforms) was moved from [Random.cpp](https://github.com/apple/swift/blob/656fe89752ae6ea673be9c4da83ee9b60b6a3aee/stdlib/public/stubs/Random.cpp#L47-L50) to [Random.h](https://github.com/apple/swift/blob/656fe89752ae6ea673be9c4da83ee9b60b6a3aee/stdlib/public/SwiftShims/Random.h#L31-L32), using `static inline SWIFT_ALWAYS_INLINE`, would it be imported into Swift as `@inlinable`?

Reply:
Yeah, we do this in a bunch of other places, and it's also (almost) what I'll do here. I'll update this sometime this week.

Reply:
You can't do that either because SwiftShims.h is not allowed to import system headers and should not declare system APIs.

Reply:
@jrose-apple We've done it in some places before (see e.g. the declarations of `lgamma_r` in `LibcShims.h`). I don't see why it wouldn't be a workable solution here as well, beyond "not wanting to have a duplicate declaration".

Reply:
It could result in those functions getting attributed to SwiftShims instead of Darwin, and if you need attributes on the function it'd be a problem if they conflict. The whole point of these trampolines is that declaring C functions in SwiftShims or in the stdlib itself has bitten us in the past.

Reply:
Is your contention that we shouldn't have any such static inline wrappers in the shims? If so, what's the alternative--just eat the performance hit for all of these operations? For random there's enough work happening that it's not catastrophic, but it's still measurable, and for some of these other operations it's worse.

Reply:
The rule is that the static ones can't call anything declared elsewhere. We've been eating the perf hit where the indirection is used for several releases; we can take a little longer (or compile with LTO) to come up with a proper solution.

Reply:
This problem is also specific to *the standard library,* because that's the only library that's below the \$Platform module.

Reply:
> The rule is that the static ones can't call anything declared elsewhere.

Except for `strlen` and `memcmp` (both of which actually have clang builtins, IIRC, so I think we could fix those) and `malloc_size` and `free` and ...

Reply:
It looks like I *was* tagged into #19614 and didn't complain then. I do not trust any of this though—the types are only matching up because the importer special-cases a few of the shims types like it does the real libc types, and the more we do it the more likely we are to accidentally diverge from the real SDK.